app/OpenGraphImage.php

```diff
@@ -25,32 +44,38 @@ public static function makeFileName($uniqueId, $name): string
         return "{$uniqueId}_" . Str::slug($name, '-') . '.png';
     }
 
-    public function __construct($fileName)
+    public function generate()
     {
-        $this->fileName = $fileName;
+
+        if (! $this->makeStorageDirectory()) {
+            $this->deleteImagesForProject();
+        }
```

Comment:
I think we can get rid of `makeStorageDirectory()` (see comment below), and just always check for other project images. So the contents of this could become:


```php
Storage::ensureDirectoryExists($this->storageDirectory());
$this->deleteImagesForProject();
$this->save($this->make());
```